client/lib/domains/index.js

```diff
@@ -1,5 +1,6 @@
 export { canCurrentUserAddEmail } from './can-current-user-add-email';
 export { canCurrentUserCreateSiteFromDomainOnly } from './can-current-user-create-site-from-domain-only';
+export { canDomainOwnerAddEmail } from './can-domain-owner-add-email';
```

Comment:
Honestly it's not clear to me what the differences between the `canCurrentUserAddEmail()` and `canDomainOwnerAddEmail()` functions are, i.e. in which case someone should pick one over the other. I would really encourage us to take the time to clarify that in the docblock comment (looking at the backend code and the eligibility check I'm not sure why we need that new function). Cross linking could also be a good idea, e.g.:

```javascript
/**
 * Determines if the current user can purchase email for the specified domain [ADD MORE CLARIFICATION]
 * 
 * @see canDomainOwnerAddEmail for checking also [ADD MORE CLARIFICATION]
 */
export function canCurrentUserAddEmail( domain: ResponseDomain | undefined ) {
	return !! domain?.currentUserCanAddEmail;
}
```

Reply:
The reason for adding `canDomainOwnerAddEmail()` is for checking if the domain owner is also the email owner (which you have helped clarified by proposing an update to the comment, thanks :)). Whereas, `canCurrentUserAddEmail()` only checks if the current user can add email. 

When the user is a domain owner but not an email owner (vice versa), the messaging [here](https://github.com/Automattic/wp-calypso/pull/68435) will not be helpful to them.

>Additional mailboxes can only be purchased by **{username}**, who is the owner of the email subscription. If you have access to that account, please [log in with the account] to make a purchase. Otherwise, please reach out to **{username}** or [contact support](https://wordpress.com/help/contact).

Log in as the email subscription owner, user will see:

>Additional mailboxes can only be purchased by **{username}**, who is the owner of **{domain}**. If you have access to that account, please [log in with the account] to make a purchase. Otherwise, please reach out to **{username}** or [contact support](https://wordpress.com/help/contact).

They will end up in an "infinite" loop, being asked to login as a different user in order to make a purchase. Therefore, we need to introduce new messaging:

>Additional mailboxes can only be purchased by the owner of the domain and email subscription. Please contact support to make a purchase.

Reply:
Thanks for the clarification @wongasy! I like Fredrik's suggestion very much. Since I'm not a native English speaker I may be wrong but I would rename it to `areDomainAndEmailSubscriptionsOwnedByDifferentUsers` though.